ttnn/cpp/ttnn/operations/index_fill/device/index_fill_multi_core_factory.cpp

```
@@ -0,0 +1,193 @@
+    } else if (std::holds_alternative<float>(fill_value)) {
+        u.f32 = std::get<float>(fill_value);
+    }
+
+    auto num_rows = input.volume() / input.get_logical_shape()[-1];
+    Program program{};
+    Device* device = input.device();
+
+    auto compute_with_storage_grid_size = device->compute_with_storage_grid_size();
+    uint32_t num_cores_y = compute_with_storage_grid_size.y;
+
+    auto [num_cores, all_cores, core_group_1, core_group_2, num_rows_per_core_group_1, num_rows_per_core_group_2] =
+        tt_metal::split_work_to_cores(compute_with_storage_grid_size, num_rows);
+
+    auto input_data_format = datatype_to_dataformat_converter(dtype);
+    auto index_data_format = datatype_to_dataformat_converter(index.get_dtype());
+    auto output_data_format = datatype_to_dataformat_converter(output.get_dtype());
+
+    uint32_t input_unit_size = input.get_logical_shape()[-1] * input.element_size();
+    uint32_t rounded_input_unit_size = round_up_to_mul32(input_unit_size);
+
+    uint32_t index_unit_size = index.volume() * index.element_size();
+    uint32_t rounded_index_unit_size = round_up_to_mul32(index_unit_size);
+
+    uint32_t output_unit_size = output.get_logical_shape()[-1] * output.element_size();
+    uint32_t rounded_output_unit_size = round_up_to_mul32(output_unit_size);
+
+    auto src_cb_index = CB::c_in0;
+    CircularBufferConfig cb_src_config =
+        CircularBufferConfig(rounded_input_unit_size, {{src_cb_index, input_data_format}})
+            .set_page_size(src_cb_index, rounded_input_unit_size);
+    auto cb_src = tt::tt_metal::CreateCircularBuffer(program, all_cores, cb_src_config);
+    std::map<string, string> reader_defines;
+
+    switch (dtype) {
+        case DataType::BFLOAT16: reader_defines["OUTPUT_DTYPE_BFLOAT16"] = "1"; break;
+        case DataType::INT32: reader_defines["OUTPUT_DTYPE_INT32"] = "1"; break;
+        case DataType::FLOAT32: reader_defines["OUTPUT_DTYPE_FLOAT32"] = "1"; break;
+        default: break;
```

If you are not going to support other types of data here, it is better to throw `TT_FATAL` to prevent a case of unexpected behaviors.

---

I think it'll be more reasonable to throw `TT_FATAL` in validating so I'll be doing that instead. Is that okay?

---

Resolved